components/operator/internal/state/registry.go

```diff
@@ -133,9 +135,45 @@ func setInternalRegistryConfig(ctx context.Context, r *reconciler, s *systemStat
 	}
 	r.log.Debugf("docker registry node port: %d", nodePort)
 	s.flagsBuilder.WithNodePort(int64(nodePort))
+
+	pvcStorage, err := resolvePVCSize(ctx, r, s)
+	if err != nil {
+		return errors.Wrap(err, "while resolving pvc size")
+	}
+	if pvcStorage != nil {
+		r.log.Debugf("docker registry pvc size: %s", pvcStorage.String())
+		s.flagsBuilder.WithRegistryPVSize(pvcStorage.String())
+	}
+
 	return nil
 }
 
+func resolvePVCSize(ctx context.Context, r *reconciler, s *systemState) (*resource.Quantity, error) {
+
+	actualStorage, err := registry.GetClaimedServerlessDockerRegistryStorageSize(ctx, r.client, s.instance.Namespace)
+	if err != nil {
+		return nil, errors.Wrap(err, "while reading actual PVC size from cluster")
+	}
+
+	r.log.Debugf("PVC size : actual %s", actualStorage.String())
+
+	if s.instance.Spec.DockerRegistry.PV != nil && s.instance.Spec.DockerRegistry.PV.Size != nil {
```

Comment:
IMO this `if` is inconsistent with x-kubernetes validation:
https://github.com/kyma-project/serverless/pull/833/files#diff-1c55a815d90f24081c70e257d769a39bdf2120359ad2db68d16b0dd8ca811e11R51

Here we check existence `PV` and `PV.Size`. In x-kubernetes validations we check only `PV`.

Reply:
No longer checking is Size is not nil - it cannot be.